internal/clients/aws.go

```diff
@@ -38,6 +39,16 @@ type SetupConfig struct {
 	Logger            logging.Logger
 }
 
+// iamRegions holds the region used for signing IAM credentials for each AWS partition.
+var iamRegions = map[string]string{
```

Comment:
What's the source of truth for this table?

Following is a more robust implementation for the signing region lookup keyed by the partition ID:
```go
	// map from a partition to its IAM signing region
	signingRegionMap := make(map[string]string)
	for _, p := range endpoints.DefaultResolver().(endpoints.EnumPartitions).Partitions() {
		for _, r := range p.Regions() {
			re, err := p.EndpointFor("iam", r.ID(), func(opt *endpoints.Options) {
				opt.ResolveUnknownService = true
			})
			if err != nil {
				panic(err)
			}
			signingRegionMap[p.ID()] = re.SigningRegion
			break
		}
	}
```

It currently produces the following map:
```
map[aws:us-east-1 aws-cn:cn-north-1 aws-iso:us-iso-east-1 aws-iso-b:us-isob-east-1 aws-iso-e:eu-isoe-west-1 aws-us-gov:us-gov-west-1]
```

Please note the difference for the partition `aws-cn`, don't know if this really matters.

Please also see the discussion below regarding partition lookups using the `region` as a key, which are also applicable here.

Also please note that we have `spec.endpoint.signingRegion` in the `ProviderConfig` API, which we should factor into this implementation. But I suggest we do it with a follow-up PR.